include/logging/log.h

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOG_H_
+#define LOG_H_
+
+#include <logging/log_instance.h>
+#include <logging/log_core.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define LOG_LEVEL_NONE  0
+#define LOG_LEVEL_ERR   1
+#define LOG_LEVEL_WARN  2
+#define LOG_LEVEL_INFO  3
+#define LOG_LEVEL_DBG   4
+
+/** @brief Module specific log level.
+ *
+ * Must be defined before including this file to override default settings.
+ *
+ * @note This header file should only be included by source files.
+ *
+ */
+#ifndef LOG_LEVEL
+#define LOG_SRC_LEVEL CONFIG_LOG_DEFAULT_LEVEL
```

Comment:
Maybe call this "LOCAL" log level or something.  "SRC" is a little confusing -- all logging comes from "source code".

Also: not a fan of "define this before including the file" preprocessor APIs. With a little care you can do something like IS_ENABLED() to test vs. symbols that may or may not be defined.

Reply:
I liked your suggestion to use something like IS_ENABLED() macro. Then we get rid of the requirement that local log level must be defined before including log.h. 

'Source' meant module or instance (source of the log). I see that this is confusing. Maybe I will come with better name.